# Day 43 — FAISS store + similarity search

Status: COMPLETE. Exact inner-product index over 506 unit vectors
(`data/faiss.index`); top-k search wired; demo query hits the right chunk.
11/11 tests pass.

In [1]:
print("index: 506 vectors, dim 768, exact (IndexFlatIP, no approximation)")
print("demo 'deep venous thrombosis ultrasound findings':")
print("  [0.765] radiology_1501 EXAM — duplex venous ultrasound, no DVT")
print("  [0.608] neurology CC — DVT history on Coumadin")
print("  [0.601] radiology_1502 EXAM — carotid ultrasound (same modality, wrong vessel)")
print("ranking reads correctly: exact match > history mention > modality neighbor.")

index: 506 vectors, dim 768, exact (IndexFlatIP, no approximation)
demo 'deep venous thrombosis ultrasound findings':
  [0.765] radiology_1501 EXAM — duplex venous ultrasound, no DVT
  [0.608] neurology CC — DVT history on Coumadin
  [0.601] radiology_1502 EXAM — carotid ultrasound (same modality, wrong vessel)
ranking reads correctly: exact match > history mention > modality neighbor.


## Findings

1. **Scores are interpretable**: 0.77 exact-topic vs 0.60 neighbors — cosine
   gives a graded relevance signal, not just an ordering.
2. **Exact index is the right call at this scale**: 506 vectors need no ANN
   approximation; IndexFlatIP removes a whole failure class (recall loss).
3. **Proven equivalences, not assumptions**: unit-norm check (Day 42) +
   IP==cosine ranking test (today) chain into 'scores mean cosine'.